routes/vendors.js

```diff
@@ -77,33 +73,37 @@ router.post('/:code/delete', function(req, res, next) {
 
 //bare bones done, more implementation needed for adding other ingredients, currently hardcoded
 router.post('/:code/add_ingredients', function(req, res, next) {
-  var ingQuery = Ingredient.getIngredient(req.body.ingredient);
-
-  ingQuery.then(function(result) {
-    console.log(result);
+  var ingQuery = Ingredient.getIngredientById(mongoose.Types.ObjectId(req.body.ingredient));
+  ingQuery.then(async function(result) {
     if (result == null) {
       let err = new Error('This ingredient does not exist');
       return next(err);
     }
-    let ingId = mongoose.Types.ObjectId(result._id);
-    VendorHelper.addIngredient(req.params.code, ingId, req.body.cost);
-    logs.makeVendorLog('Add ingredients', { 'Vendor code': req.params.code, 'Ingredient ID': ingId, 'cost': req.body.cost }, entities = ['vendor', 'ingredient'], req.session.userId);
+    await VendorHelper.addIngredient(req.params.code, result._id, req.body.cost);
```

Comment:
Why not just do `.then` here? You return a promise after all.

Reply:
i am not doing another db call so i figured if we are just waiting for a db addition before a render we didnt need the .then.   Also I kinda figured out how await works

Reply:
yeah but isn't your promise also catching and throwing errors? if you want to display those in the browser, you'll have to catch them in the route. with await you can't really do that.

Reply:
ahhh.  I see what you mean.  Yeah i can just make it a promise i guess. tbh i was just trying to save a few lines of code lol.  Don't fully understand how promises work, just how to use them lol.  Will keep that in mind